src/zenml/integrations/aws/flavors/sagemaker_step_operator_flavor.py

```diff
@@ -30,16 +30,29 @@ class SagemakerStepOperatorSettings(BaseSettings):
     """Settings for the Sagemaker step operator.
 
     Attributes:
-        instance_type: The type of the compute instance where jobs will run.
-            Check https://docs.aws.amazon.com/sagemaker/latest/dg/notebooks-available-instance-types.html
-            for a list of available instance types.
         experiment_name: The name for the experiment to which the job
             will be associated. If not provided, the job runs would be
             independent.
-    """
+        input_data_s3_uri: S3 URI where training data is located if not locally,
+            e.g. s3://my-bucket/my-data/train. How data will be made available
+            to the container is configured with estimator_args.input_mode. Two possible
+            input types:
+                - str: S3 location where training data is saved.
+                - Dict[str, str]: (ChannelName, S3Location) which represent
+                    channels (e.g. training, validation, testing) where
+                    specific parts of the data are saved in S3.
+        estimator_args: Arguments that are directly passed to the SageMaker
+            Estimator. See
+            https://sagemaker.readthedocs.io/en/stable/api/training/estimators.html#sagemaker.estimator.Estimator
+            for a full list of arguments.
+            For estimator_args.instance_type, check
+            https://docs.aws.amazon.com/sagemaker/latest/dg/notebooks-available-instance-types.html
+            for a list of available instance types.
 
-    instance_type: Optional[str] = None
+    """
```

Comment:
For backwards compatibility we would need to keep this attribute and deprecate it:
- Deprecate the attribute:
```python
from zenml.utils import deprecation_utils

class SagemakerStepOperatorSettings(BaseSettings):
  ...

  _deprecation_validator = deprecation_utils.deprecate_pydantic_attributes("instance_type")
```

Reply:
Readded `instance_type` and added the ZenML deprecation validator.